cmd/trace-agent/subcommands/run/command.go

```diff
@@ -98,13 +101,21 @@ func runTraceAgentProcess(ctx context.Context, cliParams *Params, defaultConfPat
 		}),
 		autoexitimpl.Module(),
 		statsd.Module(),
-		fx.Provide(func(coreConfig coreconfig.Component) tagger.Params {
-			if coreConfig.GetBool("apm_config.remote_tagger") {
-				return tagger.NewNodeRemoteTaggerParamsWithFallback()
-			}
-			return tagger.NewTaggerParams()
+		dualTaggerfx.Module(tagger.DualParams{
+			UseRemote: func(c coreconfig.Component) bool {
+				return c.GetBool("apm_config.remote_tagger")
+			},
+		}, tagger.Params{}, tagger.RemoteParams{
+			RemoteTarget: func(c coreconfig.Component) (string, error) {
+				return fmt.Sprintf(":%v", c.GetInt("cmd_port")), nil
+			},
+			RemoteTokenFetcher: func(c coreconfig.Component) func() (string, error) {
+				return func() (string, error) {
+					return security.FetchAuthToken(c)
+				}
+			},
+			RemoteFilter: taggerTypes.NewMatchAllFilter(),
```

Comment:
Same here.

Reply:
Even if they look almost idetical they are not, and since this is not configuration is not reused in `cmd/trace-agent` I think trying to DRY the code would make things more complicated